core/modules/ccontrol/A4UserQueryFactory.cc

```diff
@@ -34,29 +37,48 @@
 
 #include "parser/QSMySqlListener.h"
 #include "query/SelectStmt.h"
-
+#include "util/IterableFormatter.h"
 
 
 // antlr4 C++ runtime seems to require that we use namespace antlr4; trying to use it with classes
 // results in at least one error where the `tree` decl assumes that it's already operating in the
 // antlr4 namespace. I may be wrong about this though; it needs research (before merge to master).
 // If I'm right we should see about fixing the antlr4 cpp runtime & issuing a PR to them.
 using namespace antlr4;
+using namespace std;
 
 namespace {
 LOG_LOGGER _log = LOG_GET("lsst.qserv.ccontrol.A4UserQueryFactory");
+
+// For the current query, this returns a list where each pair contains a bit of the string from the query
+// and how antlr4 tokenized that bit of string. It is useful for debugging problems where antlr4 did not
+// parse a query as expected, in the case where the string was not tokenized as expected.
+vector<pair<string, string>> getTokenPairs(CommonTokenStream & tokens, QSMySqlLexer & lexer) {
+    vector<pair<string, string>> ret;
+    for (auto t : tokens.getTokens()) {
+        string name = lexer.getVocabulary().getSymbolicName(t->getType());
```

Comment:
could this be replaced with a more efficient:
auto const& name = ...

(provided the called method returns a constant reference to a string).

And the rest of the loop body could be reduced to:

ret.push_back(
  make_pair(
    name.empty() ?
      lexer.getVocabulary().getLiteralName(t->getType()) :
      name,
    t->getText()));

The later is just a matter of personal taste :-)